### ISSUE TITLE
Getting Error while Starting Sparksession

### ISSUE THREAD

**Issue by Ahmetemintek** [huggingface/huggingface_hub#502]
Hello, I am trying to create new streamlit app on HuggingFace by using pyspark in it. <br/>
I've created app.py file, requirement.txt file and I run a basic streamlit app without pyspark and it worked seamlessly. <br/>
However, problem starts when I add the following line `spark = SparkSession.builder.appName("appName").getOrCreate()` to start sparksession, I got an error like following: <br/>
` Exception: Java gateway process exited before sending its port number`

The content of requirements.txt file: <br/>
```
streamlit
pyspark==3.1.2
spark-nlp==3.3.2
```
If Anyone worked with pyspark on hf can help me I would be appreciate!

**Comment by julien-c** [comment]
What do you usually need as system dependencies to run this?

**Comment by Ahmetemintek** [comment]
On my local, I have streamlit, pyspark and spark-nlp installed, to start SparkSession I have jdk version 10.0.1 . And my apps run seamlessly. However, in Huggingface space even though I've defined all dependencies in requirements.txt file, it returns java error. I don't know what should I do for jdk for HF space.

**Comment by Ahmetemintek** [comment]
To clarify the issue, let me explain step by step. <br/>
The content of my app like following:

```
import streamlit as st
import sparknlp
from sparknlp.base import *
from sparknlp.annotator import *
from pyspark.sql import functions as F
from pyspark.sql.functions import col, when, count, isnan
from pyspark.ml import Pipeline

st.write("Sparknlp Version: " + sparknlp.version())

```
My streamlit app works seamlessly so far and types the version of sparknlp properly.  However, when I start SparkSession like following: 

`spark= sparknlp.start()`
or like: 
`ss = SparkSession.builder.appName('first_draft').getOrCreate()
`
app returns the error as `Exception: Java gateway process exited before sending its port number`

content of my requirements.txt is still like following: 
```
streamlit
pyspark==3.1.2
spark-nlp==3.3.2
```

**Comment by julien-c** [comment]
yes, the base image for Spaces does not include Java, but you can add system dependencies through a `packages.txt` (cc @cbensimon @nateraw)

See here for instance `default-jdk` to install Java: https://huggingface.co/spaces/elonmuskceo/sparknlp/blob/main/packages.txt (I don't remember if it's documented but base image for Spaces is currently Debian Buster, cc @osanseviero)

Then I can successfully instantiate a `sparknlp.start()` object, **see resulting space: https://huggingface.co/spaces/elonmuskceo/sparknlp**

Hope this helps. Out of curiosity can you share your Space here?
